### ISSUE TITLE
Rounded corners of Windows 11 windows should be transparent

### ISSUE THREAD

**Issue by bllngr** [greenshot/greenshot#326]
**Describe the bug**
The new Windows 11 features rounded corners for window decorations, which are captured with a gray background by Greenshot.

**To Reproduce**
Example steps to reproduce the behavior:
1. Install the Windows 11 preview
2. Install Greenshot
3. Capture an arbitraty window, and open it in the image editor
4. Observe the rounded corners having a gray background

**Expected behavior**
Rounded corners should have a transparent background instead.

**Screenshots**
![image](https://user-images.githubusercontent.com/303458/126761105-6043d93d-c1bd-495d-b293-0189225f3bf7.png)

**Versions (please complete the following information):**
 - Greenshot version 1.2.10 Build 6
 - Microsoft Windows 11 Pro Version 10.0.22000 Build 22000

**Comment by Lakritzator** [comment]
Yes, I agree, but we still have time to tackle this. I need to see how I will test Greenshot on Windows 11 yet.

**Comment by Lakritzator** [comment]
btw... rounded corners are a nightmare for screenshot tools... especially if they are anti-aliased, currently I don't have access to the pixels before they are blended, after they are I cannot 100% get a perfect screenshot. I am looking into this...

**Comment by abdulhaquemasum** [comment]
[7capture](https://www.7capture.com/) seems to know how to do it. I've been a Greenshot user since 2015, would love this feature please.

**Comment by Lakritzator** [comment]
@abdulhaquemasum Greenshot was already able to do this for Windows 7, I would say much longer than 2015, so you must have seen this 😉 Anyway for Windows 11, it just worked differently... 

I finally found a solution, and hope I have integrated this correctly, but this is where I need some people to test this.
This can be tested with version 1.3.238 (or later) which can be found here: https://github.com/greenshot/greenshot/releases

**Comment by esvres** [comment]
I've been testing Greenshot 1.3.238 on Windows 11 22000.469, and these are the issues I have encountered: 

- If the window is in focus, and I use the "Capture Window" hotkey, the window is cropped on the right, with a black bar on the left and black corners. [Screenshot](https://i.imgur.com/jwFnoEr.png).
- If the window is not in focus, and I select "Capture Window" from the tray icon, the window is fully visible, but corners are not transparent. [Screenshot](https://i.imgur.com/HLiJg7Y.png).
- If the window is maximized, and I use the "Capture Window" hotkey, the window is cropped on the right. [Screenshot](https://i.imgur.com/lbgaGr6.png).
- If the window is maximized, and I select "Capture Window" from the tray icon, the window is cropped on both the left and the right. [Screenshot](https://i.imgur.com/eM0QhxU.png). This can be also seen in the window selection box. [Screenshot](https://i.imgur.com/D8bYAGa.png).

**Comment by Lakritzator** [comment]
Thanks for the thorough tests, this more or less reflects the behavior I saw on Windows 10 but on Windows 11 I didn't see any problems yet. This is why I didn't use this technology for Windows 10. That is more or less pretty bad news.

I have some questions:

1. Do you have multiple monitors?
2. If 1. is yes, are they with different DPI settings? (100%, 125%, 150%?)
3. Did you try the capture on a focused window which is completely on the screen?
4. Did you open them in the editor? Did you see a "checkerbox" in the background, indicating that there is transparency?

What I saw on Windows 10, I can only guess why this is, is when the window is completely visible on the screen everything worked ok, but if a part was not on the screen things started to work badly, like you noticed.
I really was hoping it would work as I saw on my Windows 11 installation 😢 

I'm not sure if I can do that much, but there is a bit I can change on the routine... let's see where it goes.

**Comment by esvres** [comment]
Hello,

1. Yes, but my display configuration was set to "Second screen only" during testing, so it was effectively one monitor (1440p, 100% scaling). To be on the safe side, I repeated the tests after unplugging my secondary monitor and rebooting, while only using the primary one (1080p, 125% scaling): the results remained unchanged.
2. //
3. Yes; see the first and the third test. I couldn't keep the window focused in the second and fourth test, as right clicking the tray icon in order to select the "Capture Window" option makes the window lose focus. In any case, the window was on the screen and fully visible.
4. Yes; in neither scenario was the checkerbox present. [Screenshot](https://i.imgur.com/xyMtNLV.png).

I'm glad I could be of some help.

**Comment by abdulhaquemasum** [comment]
Hi @Lakritzator thanks for looking into this, did you manage to get to the bottom of the issue? I haven't upgraded to Windows 11 yet, this is the only app stopping me :)

**Comment by Void48** [comment]
Any update on this issue?

**Comment by Slapbox** [comment]
Windows 11's inbuilt Alt+PrintScreen has the same gnarly edges, so this is hardly an issue limited to Greenshot.
